biojava-structure-gui/src/main/java/org/biojava/nbio/structure/align/gui/jmol/JmolPanel.java

```diff
@@ -114,11 +127,28 @@ public void executeCmd(String rasmolScript) {
 		viewer.evalString(rasmolScript);
 	}
 
-	public void setStructure(Structure s)
+	public void setStructure(final Structure s)
 	{
 		this.structure = s;
-		String serialized = s.toMMCIF();
-		viewer.openStringInline(serialized);
+		try (
+				PipedOutputStream out = new PipedOutputStream();
+				// Viewer requires a BufferedInputStream for reflection
+				InputStream in = new BufferedInputStream(new PipedInputStream(out));
+				) {
+			new Thread((Runnable)() -> {
+				try {
+					MmtfActions.writeToOutputStream(s,out);
+				} catch (Exception e) {
+					logger.error("Error generating MMTF output for {}",
+							s.getStructureIdentifier()==null ? s.getStructureIdentifier().getIdentifier() : s.getName(), e);
+				}
+			}).start();
+			viewer.openReader(null, in);
+		} catch (IOException e) {
```

Comment:
How about throwing the exception? In the end of the day this will only produce a blank jmol screen (and a message in the logs), but a user will not be able to know if there was a problem in the loading. If we throw it then a user can catch it down the line and act upon it.